README.md

```diff
@@ -19,12 +19,12 @@ require 'prometheus/client'
 prometheus = Prometheus::Client.registry
 
 # create a new counter metric
-http_requests = Prometheus::Client::Counter.new(:http_requests, 'A counter of HTTP requests made')
+http_requests = Prometheus::Client::Counter.new(:http_requests, docstring: 'A counter of HTTP requests made')
```

Comment:
I'm generally with you on the "not breaking the contract". This is, for example, why I didn't change the namespaces, even though I'd prefer a "flatter" namespace structure, and i've seen comments around about that too (can't find one right now, though).
That seemed like "unnecessarily" breaking the contract.

However, we're breaking the contract on other places, mainly, demanding that labels be declared up-front when a new metric is instantiated. I consider that a "good" break of the contract, since it takes us more in line with best practices.

`docstring` is the one i'm most on the fence about.
I don't fully like it, but I also don't like the two nakes strings (metric_name and docstring). I feel like the first one is pretty self-explanatory, but the second one not so much.

That said, if you want me to make docstring a non-kwarg, I'll happily do so.

Reply:
I figured out later that API is changing, no big deal. It is used widely in the library anyway.